### DISCUSSION TITLE
JWT how to customize Scheme_name?

### DISCUSSION THREAD

**Discussion by wxq0309** [fastapi/fastapi#7639] [Questions] [answered]
Hello, I recently used the Fastapi framework and wanted to use “ TOKEN + jwt_string” for JWT verification. I set scheme_name = TOKEN in OAuth2PasswordBearer, but I directly judged in the source code of OAuth2PasswordBearer whether scheme_name.lower () is equal to " "Bearer", how can this be resolved?

**Comment by tiangolo** [comment]
I assume you solved your problem, thanks for closing the issue :+1:

**Comment by YuriiMotov** [comment, marked as answer]
`scheme_name` parameter of `OAuth2PasswordBearer` is just the name of this particular security requirement object in openapi scheme. This is the name that will be used to reference to this scheme from the protected path operations.

This is not the scheme that is used in `Authorization` header.
`Bearer` in `Authorization` header is defined by OAuth2 standard and can't be changed.
If you want to build your custom authorization scheme, subclass from `SecurityBase` and implement the logic you need.
Remember, that you will not be able to describe your custom security scheme in openapi (it doesn't support custom security schemes)
